custom/enikshay/tasks.py

```diff
@@ -638,3 +582,63 @@ def _get_diagnostic_test_name(self, diagnostic_test):
                 diagnostic_test.get_case_property('investigation_type_name'), site_specimen_name)
         else:
             return diagnostic_test.get_case_property('investigation_type_name')
+
+
+def calculate_dose_status_by_day(adherence_cases):
+    """
+    adherence_cases: list of 'adherence' case dicts that come from elasticsearch
+    Returns: {day: DoseStatus(taken, missed, unknown, source)}
+    """
+
+    adherence_cases_by_date = defaultdict(list)
+    for case in adherence_cases:
+        adherence_date = parse_date(case['adherence_date']) or parse_datetime(case['adherence_date']).date()
+        adherence_cases_by_date[adherence_date].append(case)
+
+    status_by_day = defaultdict(lambda: DoseStatus(taken=False, missed=False, unknown=True, source=False))
+    for d, cases in adherence_cases_by_date.iteritems():
```

Comment:
is `d` day here? Having such names does cause trouble during debugging. There are time where i use ipdb to debug something but can't check the value of a variable because it means something else for the debugger like 'c' for continue. Would be good to keep this actually a name for clarity as well.